torch/nn/_functions/thnn/upsampling.py

```diff
@@ -6,10 +6,10 @@
 from . import _all_functions
 
 
-class _UpsamplingBase(Function):
-
+class UpsamplingNearest2d(Function):
+    
     def __init__(self, size=None, scale_factor=None):
-        super(_UpsamplingBase, self).__init__()
+        super(UpsamplingNearest2d, self).__init__()
```

Comment:
I think leaving the base class makes sense, as we could allow tuple scale_factors for upsampling nearest too. Then, a subclass constructor would check that the factors are integral and equal.